unlock-app/src/vendor.d.ts

```diff
@@ -1,7 +1,50 @@
 // This file is meant to declare modules we depend on which do not have @types
 // packages available. Defining them here will unblock us, though doing that the
 // most convenient way will not buy us any type safety.
+import 'react-stripe-elements'
+
 declare module 'react-jazzicon'
 
 // TODO: Remove this when we have a TS build of unlock-js
 declare module '@unlock-protocol/unlock-js'
+
+type PaymentMethodType = 'card' | 'card_present'
+
+// vastly simplified from the real types, but this is all we should need
+interface PaymentMethodIncomplete {
+  billing_details?: {
```

Comment:
do all of these need to be ?: or are there a subset of possibilities that we expect? maybe we could do a union type of those subsets?

Reply:
the `billing_details` implementation here was pulled from the Stripe @types file. These are all optional properties on the object in their API (actually, almost everything is an optional object property). We could possibly restrict it to just the subset of properties we expect to have (or care about -- in the end we only need to get the `id` from `PaymentMethod`), but at some point in the not-too-distant future we should be deleting this code and just pulling in from DefinitelyTyped.